### ISSUE TITLE
Tool Script: Notified On Save?

### ISSUE THREAD

**Issue by hasahmed** [godotengine/godot#27342]
**Godot version:**

3.1

I am attempting to write a script for a node, that I would like to save a resource to disk every time the scene is saved. The scripts overrides the `_notification` function, and it appears that there are 3 notifications emitted in the following order:
`NOTIFICATION_PARENTED` : 18 (from Node)
`NOTIFICATION_UNPARENTED` : 19 (from Node)
`NOTIFICATION_PREDELETE` : 1 (from Object)
(as to why those are emitted I have no idea, but thats a separate issue)

Anyway attempting to run my code my script when those notifications occur cause me errors. (I think it has to do with the state of the tree, and what I am trying to do)

Long story short, I would like a notification that I can hook into for specifically when a save happens, and while the tree is still valid.

If there is a way to make this happen that I have not come across, then please enlighten me.

**Comment by santouits** [comment]
Hi, you can do this by making an editor plugin and override the save_external_data method. Read here http://docs.godotengine.org/en/latest/tutorials/plugins/editor/making_plugins.html:

```
tool extends EditorPlugin

func save_external_data():
	print("save")
```

Though I reproduced the notifications emitted whie saving the project and they seem like a bug.

**Comment by Zylann** [comment]
> I would like to save a resource to disk every time the scene is saved

If the point is to save a custom resource, why not make a custom resource? They directly cover that use case. If the resource is stored in a property of the node with the `STORAGE` hint, Godot will save it alongside the scene. And if that resource has an associated file, it will save to that file.

**Comment by KoBeWi** [comment]
There is `NOTIFICATION_EDITOR_PRE_SAVE` and `NOTIFICATION_EDITOR_POST_SAVE` now.

**Comment by Adomo-Software** [comment]
> There is `NOTIFICATION_EDITOR_PRE_SAVE` and `NOTIFICATION_EDITOR_POST_SAVE` now.

No, there is not.
